osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SystemEventServiceTest.java

```diff
@@ -0,0 +1,151 @@
+            this.osgpRequestMessageSender,
+            this.correlationIdProviderService,
+            this.invocationCounterEventThreshold);
+  }
+
+  @Test
+  void verifyMaxValueNotReachedEvent() {
+    final DlmsDevice device =
+        new DlmsDeviceBuilder()
+            .withInvocationCounter(this.invocationCounterEventThreshold - 1)
+            .build();
+
+    final MessageMetadata messageMetadata =
+        new Builder().withOrganisationIdentification("org-id").build();
+
+    this.service.verifySystemEventThresholdReachedEvent(device, messageMetadata);
+
+    verifyNoInteractions(this.correlationIdProviderService);
+    verify(this.osgpRequestMessageSender, never())
+        .send(
+            any(RequestMessage.class),
+            eq(MessageType.SYSTEM_EVENT.name()),
+            any(MessageMetadata.class));
+  }
+
+  @Test
+  void verifyInvocationCounterIsNull() {
+    final DlmsDevice device = new DlmsDeviceBuilder().withInvocationCounter(null).build();
+
+    final MessageMetadata messageMetadata =
+        new Builder().withOrganisationIdentification("org-id").build();
+
+    this.service.verifySystemEventThresholdReachedEvent(device, messageMetadata);
+
+    verifyNoInteractions(this.correlationIdProviderService);
+    verify(this.osgpRequestMessageSender, never())
+        .send(
+            any(RequestMessage.class),
+            eq(MessageType.SYSTEM_EVENT.name()),
+            any(MessageMetadata.class));
+  }
+
+  @Test
+  void verifyMaxValueReachedEvent() {
+    final DlmsDevice device =
+        new DlmsDeviceBuilder()
+            .withDeviceIdentification("device-1")
+            .withInvocationCounter(this.invocationCounterEventThreshold)
+            .build();
+
+    final MessageMetadata messageMetadata =
+        new Builder()
+            .withOrganisationIdentification("org-id")
+            .withDomain("domain")
+            .withDomainVersion("1.0")
+            .build();
+
+    when(this.correlationIdProviderService.getCorrelationId(
+            messageMetadata.getOrganisationIdentification(), device.getDeviceIdentification()))
+        .thenReturn("corr-id");
+
+    this.service.verifySystemEventThresholdReachedEvent(device, messageMetadata);
+
+    final ArgumentCaptor<MessageMetadata> messageMetadataCaptor =
+        ArgumentCaptor.forClass(MessageMetadata.class);
+    final ArgumentCaptor<RequestMessage> requestMessageCaptor =
+        ArgumentCaptor.forClass(RequestMessage.class);
+
+    verify(this.osgpRequestMessageSender)
+        .send(
+            requestMessageCaptor.capture(),
+            eq(MessageType.SYSTEM_EVENT.name()),
+            messageMetadataCaptor.capture());
+
+    final RequestMessage requestMessage = requestMessageCaptor.getValue();
+    assertThat(requestMessage.getDeviceIdentification())
+        .isEqualTo(device.getDeviceIdentification());
+    assertThat(requestMessage.getDeviceIdentification())
+        .isEqualTo(device.getDeviceIdentification());
```

Comment:
Asserting the device identification once should be enough.

Reply:
Changed